### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by MEDATTA0**
userRoutes and todoRoutes are more structured. The SQL queries are replaced from connection.query() to connection.execute(), which is more secure.

### CODE REVIEW

File: routes/usersRoutes.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+router.get("/:userId", async (req, res) => {});
+
+//creating user
+router.post("/", async (req, res) => {});
+
+//Patch username
+router.patch("/:userId/username", async (req, res) => {});
+
+//Delete password
+router.delete("/:userId/password", (req, res) => {});
```
Comment (AbdourahamaneIssakaSani) [inline routes/usersRoutes.js:22]: Can someone delete a password ?

File: routes/usersRoutes.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+//Modify password
+router.patch("/:userId/password", async (req, res) => {});
+
+//The route for todos operations of a user
+const todoRoutes = require("./todosRoutes");
+router.use("/:userId/todos", todoRoutes);
```
Comment (AbdourahamaneIssakaSani) [inline routes/usersRoutes.js:29]: Good one, I love this

File: routes/usersRoutes.js (hunk 1 above)
Comment (MEDATTA0) [reply routes/usersRoutes.js:22]: Ohh, sorry I would like to say when you want to reset your password.

Comment (AbdourahamaneIssakaSani) [reply routes/usersRoutes.js:22]: > Ohh, sorry I would like to say when you want to reset your password.

okay, You can make it `POST /reset`.

File: controllers/todosController.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  try {
+    response = await todosModels.createTodo(
+      userId,
+      title,
+      task_description,
+      isCompleted
+    );
+    res.status(201).json({
+      message: "Task added successfully",
+      taskId: result.task_id,
```
Comment (AbdourahamaneIssakaSani) [inline controllers/todosController.js:57]: By convention
```suggestion
      data: {
      taskId: result.task_id,
      },
```

Comment (MEDATTA0) [reply controllers/todosController.js:57]: It's fixed.

File: routes/usersRoutes.js (hunk 1 above)
Comment (MEDATTA0) [reply routes/usersRoutes.js:22]: Okay

**Pull request merged**